src/ArrayInterface.jl

```diff
@@ -35,6 +35,8 @@ Otherwise, return `nothing`.
 known_length(x) = known_length(typeof(x))
 known_length(::Type{<:NTuple{N,<:Any}}) where {N} = N
 known_length(::Type{<:NamedTuple{L}}) where {L} = length(L)
+known_length(::Type{T}) where {T<:Base.Slice} = known_length(parent_type(T))
+known_length(::Type{<:Tuple{Vararg{Any,N}}}) where {N} = N
```

Comment:
I'll remove the `NTuple` method.

Reply:
If we don't have this defined for tuples then the method would return `nothing`, implying the length isn't known.

Reply:
The `Tuple{Vararg{Any,N}}` definition works for both homogenous and heterogenous tuples, so the `NTuple`-specific definition was unnecessary.
```julia
julia> NTuple{1423} <: Tuple{Vararg{Any,1423}}
true
```

Reply:
I misunderstood. I thought the final idea was to remove both `Tuple` methods, sorry.

Reply:
To expand a little on Chris' answer, in case you didn't know:
```julia
julia> (1, 'd') isa NTuple{2,<:Any}
false

julia> (1, 'd') isa NTuple{2,Any}
true
```